src/sentry/integrations/pipeline.py

```diff
@@ -39,8 +39,11 @@ def ensure_integration(key, data):
         defaults=defaults
     )
     if not created:
+        default_identity_id = integration.metadata.get('default_identity_id')
         integration.update(**defaults)
-
+        if default_identity_id:
+            integration.metadata['default_identity_id'] = default_identity_id
+            integration.update(metadata=integration.metadata)
```

Comment:
@EvanPurkhiser I think you mentioned you wanted the first identity associated with the integration not the last? If not I'll remove this bit of code.

Reply:
Actually I don't think we'll need to do this.

What I think we would want to do is attach the created identity to the created `OrganizationIntegration` object.

So you could actually move down the line

```py
org_integration_args = {}

if self.needs_default_identity:
    if not identity_model:
        raise NotImplementedError('Integration requires an identity')
    org_integration_args = {'default_auth_id': identity_model.id}

org_integration = integration.add_organization(self.organization.id, **org_integration_args)
```

Then when instantiating the "installed" integration we will probably have to do something like pass in the organization so that it can lookup the `OrganizationIntegration` which has the associated identity with access token.